pkg/healthchecker/health_checker.go

```diff
@@ -168,3 +164,66 @@ func execCommand(timeout time.Duration, command string, args ...string) (string,
 	}
 	return strings.TrimSuffix(string(out), "\n"), nil
 }
+
+// kubeletHttpHealthCheck checks the health api response on kubelet.
+// Returns true for healthy, false otherwise.
+func kubeletHttpHealthCheck(healthCheckTimeout time.Duration) bool {
+	httpClient := http.Client{Timeout: healthCheckTimeout}
+	response, err := httpClient.Get(types.KubeletHealthCheckEndpoint)
+	if err != nil || response.StatusCode != http.StatusOK {
+		glog.Info("kubelet failed http health check")
+		return false
+	}
+	return true
+}
+
+// kubeletConnectionHealthCheck checks for the kubelet-apiserver connection issue
+// by checking repeated occurrences of log "use of closed network connection" in kubelet logs.
+// Returns true if the pattern does not exist 10 times since start of service or the last 10 min, false otherwise.
+func kubeletConnectionHealthCheck() (bool, error) {
+	kubeletUptimeFunc := getUptimeFunc(types.KubeletComponent)
+	uptime, err := kubeletUptimeFunc()
+	if err != nil {
+		return true, err
+	}
+	logStartTime := time.Now().Add(-uptime).Format(types.LogParsingTimeLayout)
+	if err != nil {
+		return true, err
+	}
+	out, err := execCommand(types.CmdTimeout, "/bin/sh", "-c",
+		// Query kubelet logs since the logStartTime
+		`journalctl --unit kubelet --since "`+logStartTime+
```

Comment:
does this assume the presence of this binary ? Is it running this from the container or using hostPath ?